packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -148,30 +218,47 @@ export const useArbTokenBridge = (
   const walletAddressCached = useCallback(async () => {
     if (walletAddress) {
       return walletAddress
-    } else {
-      const address = await bridge.l1Bridge.getWalletAddress()
-      setWalletAddress(address)
-      return address
     }
-  }, [walletAddress, bridge])
 
-  const depositEth = async (weiValue: BigNumber) => {
-    const etherVal = utils.formatUnits(weiValue, 18)
-    const tx = await bridge.depositETH(weiValue)
+    const address = await l1.signer.getAddress()
+    setWalletAddress(address)
+    return address
+  }, [walletAddress, l1.signer])
+
+  const depositEth = async (amount: BigNumber) => {
+    if (typeof l2.signer.provider === 'undefined') {
+      throw new Error(`No provider found for L2 Signer`)
+    }
```

Comment:
Happy to update this error message if needed.

Reply:
seems chill